service/src/main/java/uk/gov/hmcts/reform/fpl/service/orders/validator/EPOEndDateValidator.java

```diff
@@ -0,0 +1,69 @@
+package uk.gov.hmcts.reform.fpl.service.orders.validator;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.order.OrderQuestionBlock;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+
+import static uk.gov.hmcts.reform.fpl.model.order.OrderQuestionBlock.EPO_EXPIRY_DATE;
+
+@Component
+@RequiredArgsConstructor(onConstructor_ = {@Autowired})
+public class EPOEndDateValidator implements QuestionBlockOrderValidator {
+
+    private static final String INVALID_TIME_MESSAGE = "Enter a valid time";
+    private static final String FUTURE_DATE_MESSAGE = "Enter an end date in the future";
+    private static final String END_DATE_RANGE_MESSAGE = "Emergency protection orders cannot last longer than 8 days";
+
+    private static final LocalTime MIDNIGHT = LocalTime.of(0, 0, 0);
```

Comment:
There is already a constant in the LocalTime class for this, called the exact same thing 😆